python/ml4ir/base/pipeline.py

```diff
@@ -146,6 +149,24 @@ def __init__(self, args: Namespace):
         feature_config_dict = self.file_io.read_yaml(args.feature_config)
         model_config_dict = self.file_io.read_yaml(args.model_config)
 
+        # Adding fr as a trainable feature to learn the positional bias according to MLranker.
+        if 'positional_bias_handler' in model_config_dict and model_config_dict['positional_bias_handler']['key'] == PositionalBiasHandler.MLRANKER:
+            fr_feature = \
+            {
+              'name': 'fr',
+              'node_name': 'fr_feature',
+              'trainable': True,
+              'dtype': 'int64',
+              'log_at_inference': True,
+              'feature_layer_info': {'type': 'numeric', 'shape': None},
+              'serving_info': {'name': 'fr_feature', 'required': True},
+              'tfrecord_type': 'context',
+              'preprocessing_info': [{'fn': 'convert_fr_to_one_hot','args': {'max_ranks': 25, 'mask_fr': True}}]}
+            feature_config_dict['features'].append(fr_feature)
+
+
+
+
```

Comment:
@mohazahran Why do you need this? I am very confused by the `fr_feature` feature. Why not dynamically compute it during activation/loss computation?